content/en/admin/troubleshooting.md

```diff
@@ -30,3 +30,13 @@ Check that you are specifying the correct environment with `RAILS_ENV=production
 ## **I encountered a compilation error while executing `RAILS_ENV=production bundle exec rails assets:precompile`, but no more information is given. How to fix it?**
 
 Usually it's because your server ran out of memory while compiling assets. Use a swapfile or increase the swap space to increase the memory capacity. Run `RAILS_ENV=production bundle exec rake tmp:cache:clear` to clear cache, then execute `RAILS_ENV=production bundle exec rails assets:precompile` to compile again. Make sure you clear the cache after a compilation error, or it will show "Everything's OK" but leave the assets unchanged.
+
+## **I am getting this error: `Read-only file system @ dir_s_mkdir`. Why?**
+
+Currently mastodon uses hardening sandbox mode implemented by SystemD. More details with the sandboxing options to get acquainted here - https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Sandboxing.
+
+By default, Mastodon makes use of [systemd's sandboxing capabilities](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Sandboxing) in a way that disallows writing outside of `/home/mastodon`. If Mastodon is installed elsewhere, you may need to `allow mastodon-sidekiq` and `mastodon-web` to write to a custom directory:
```

Comment:
is `allow mastodon-sidekiq` a command? I don't think it is, since the sandboxing configuration is stored in the systemd file. Since `allow` isn't a command for the user to run, it's confusing to enclose it in backticks.

```suggestion
By default, Mastodon makes use of [systemd's sandboxing capabilities](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Sandboxing) in a way that disallows writing outside of `/home/mastodon`. If Mastodon is installed elsewhere, you may need to allow `mastodon-sidekiq` and `mastodon-web` to write to a custom directory:
```